### PR TITLE
Remove userCanAdd from Job

### LINKED ISSUE #1130: [Code health] Remove “Contributors can add” settings
Remove from code and UI; contributions can no longer add LOIs. They will eventually be able to add geometries as part of a task, but these will be kept separate from LOIs.

### PR DESCRIPTION
**Pull request by JSunde**
Closes #1130.

### CODE REVIEW

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java [hunk 1]
```diff
@@ -70,11 +69,10 @@ public void testActivateSurvey_managersCanAddFeaturesToAllJobs() {
 
     surveyRepository.activateSurvey("id");
 
-    Job expectedJob = job.toBuilder().setUserCanAdd(LocationOfInterestType.ALL).build();
     surveyRepository
         .getActiveSurvey()
         .test()
-        .assertValue(p -> p.get().getJobs().equals(ImmutableList.of(expectedJob)));
+        .assertValue(p -> p.get().getJobs().equals(ImmutableList.of(job)));
   }
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java:76]: Do we even need these test anymore? Earlier it was asserting that the inserted job was modified based on the user role. Now we are checking that the job is unmodified.

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java [hunk 2]
```diff
@@ -85,11 +83,10 @@ public void testActivateSurvey_ownersCanAddFeaturesToAllJobs() {
...
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java:90]: Same as above

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java [hunk 3]
```diff
@@ -100,11 +97,10 @@ public void testActivateSurvey_contributorsCannotAddFeaturesToAnyJobs() {
 
     surveyRepository.activateSurvey("id");
 
-    Job expectedJob = job.toBuilder().setUserCanAdd(ImmutableList.of()).build();
     surveyRepository
         .getActiveSurvey()
         .test()
-        .assertValue(p -> p.get().getJobs().equals(ImmutableList.of(expectedJob)));
+        .assertValue(p -> p.get().getJobs().equals(ImmutableList.of(job)));
   }
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java:104]: Same as above

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java (hunk 1 above)
Comment (JSunde) [reply ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.java]: Good catch! Removed!

**Review by gino-m — APPROVED** [review]
LGTM! Thanks for the cleanup!

**Pull request merged**